### PR TITLE
Add difficulty adjustment settings to editor setup screen

### PR DESCRIPTION
**Pull request by peppy**
- [x] Depends on #10389
- [x] Depends on #10390

### CODE REVIEW

**Review by smoogipoo — DISMISSED** [review]

**Comment by Apollo-P** [comment]
Any thoughts on allowing circle sizes 0-2 and 7-10 to be accessed through this panel, now that such settings are allowed into the ranked section?

**Comment by enoslayd** [comment]
Is it intentional (or for now) that non existing editor settings for a ruleset (ex. Ar and cs on mania) appears?

**Comment by peppy** [comment]
Per-ruleset specifications will be added much later. I'm not about to hack these in place, and doing it properly takes attention to detail (that can come later).

File: osu.Game/Screens/Edit/Setup/DifficultySection.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+                healthDrainSlider = new LabelledSliderBar<float>
+                {
+                    Label = "Health Drain",
+                    Description = "The rate of passive health drain throughout playable time",
+                    Current = new BindableFloat(Beatmap.Value.BeatmapInfo.BaseDifficulty.DrainRate)
+                    {
+                        Default = BeatmapDifficulty.DEFAULT_DIFFICULTY,
+                        MinValue = 0,
+                        MaxValue = 10
+                    }
```
[the remark this answers is not in the document]
Comment (smoogipoo) [reply osu.Game/Screens/Edit/Setup/DifficultySection.cs:53]: It really should have a sane precision specified, even if it's just 0.1.

[the remark this answers is not in the document]
Comment (peppy) [reply osu.Game/Screens/Edit/Setup/DifficultySection.cs:53]: Will add 0.1 for now. The "hold shift" precision will be added in a future pass of the `OsuSliderBar` implementation.

**Comment by smoogipoo** [comment]
> Any thoughts on allowing circle sizes 0-2 and 7-10 to be accessed through this panel

Can be done as a separate PR.

**Pull request merged**